### PR TITLE
Initial QMK Support for the Han60

### PR DESCRIPTION
**Pull request by farhandsome**
## Description

Adding QMK support for the upcoming keyboard Han60.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* N/A

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/han60/rules.mk [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (drashna) [inline keyboards/han60/rules.mk:21]:
```suggestion
```

**Review by drashna — APPROVED** [review]

File: keyboards/han60/config.h [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0xFEED
+#define PRODUCT_ID   0xHB60
```
Comment (drashna) [inline keyboards/han60/config.h:24]: This needs to be a hex value.

Comment (farhandsome) [reply keyboards/han60/config.h:24]: Is the vendor ID fine? Or both need to be changed?

**Review by drashna — APPROVED** [review]

File: keyboards/han60/info.json [hunk 3]
```diff
@@ -0,0 +1,355 @@
+{
+    "keyboard_name": "han60",
+    "maintainer": "qmk",
+    "url": "hanboards.com",
+    "layouts": 
+	{
+        "LAYOUT_all":
```
Comment (zvecr) [inline keyboards/han60/info.json:7]: None of the layouts defined here match the content of `han60.h`

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/han60/info.json (hunk 3 above)
Comment (farhandsome) [reply keyboards/han60/info.json:7]: The `layout_all` should match the format of the `.h` file? Or do I need to change `han60.h` to correspond to each layout listed in info.json?

File: keyboards/han60/readme.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# han60
+
+![han60](https://i.imgur.com/nfiSGNO.jpg)
```
Comment (fauxpark) [inline keyboards/han60/readme.md:3]:
```suggestion
![han60](https://i.imgur.com/nfiSGNOh.jpg)

```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/han60/readme.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
+# han60
```
Comment (fauxpark) [inline keyboards/han60/readme.md:1]: Please use Unix line endings for all files.

File: keyboards/han60/keymaps/default/readme.md [hunk 6]
```diff
@@ -0,0 +1 @@
+# The default keymap for han60
```
Comment (fauxpark) [inline keyboards/han60/keymaps/default/readme.md:1]: As above.

File: keyboards/han60/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,35 @@
+/* Copyright 2021 farhandsome
```
Comment (fauxpark) [inline keyboards/han60/keymaps/default/keymap.c:1]: As above.

File: keyboards/han60/info.json [hunk 8]
```diff
@@ -0,0 +1,355 @@
+{
```
Comment (fauxpark) [inline keyboards/han60/info.json:1]: As above.

File: keyboards/han60/han60.h [hunk 9]
```diff
@@ -0,0 +1,41 @@
+/* Copyright 2021 farhandsome
```
Comment (fauxpark) [inline keyboards/han60/han60.h:1]: As above.

File: keyboards/han60/han60.c [hunk 10]
```diff
@@ -0,0 +1,17 @@
+/* Copyright 2021 farhandsome
```
Comment (fauxpark) [inline keyboards/han60/han60.c:1]: As above.

File: keyboards/han60/config.h [hunk 11]
```diff
@@ -0,0 +1,152 @@
+/*
```
Comment (fauxpark) [inline keyboards/han60/config.h:1]: As above.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/han60/readme.md (hunk 5 above)
Comment (farhandsome) [reply keyboards/han60/readme.md:1]: Do you mean `\n` rather than `\`?

Comment (fauxpark) [reply keyboards/han60/readme.md:1]: LF rather than CRLF.

File: keyboards/han60/readme.md [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+
+![han60](https://i.imgur.com/nfiSGNOh.jpg)
+
```
Comment (fauxpark) [inline keyboards/han60/readme.md:4]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/han60/han60.h [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+	K41, K42, K43,                K44,                K47, K48, K4A, K4C  \
+) { \
+    { K00, K01, K02, K03, K04, K05, K06, K07, K08, K09, K0A, K0B, K0C, K0D }, \
+    { K10, K11, K12, K13, K14, K15, K16, K17, K18, K19, K1A, K1B, K1C, K1D }, \
+	{ K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K2A, K2B, K2C, K2D }, \
+	{ K30, K31, K32, K33, K34, K35, K36, K37, K38, K39, XXX, K3B, XXX, K3D }, \
+	{ XXX, K41, K42, K43, K44, XXX, XXX, K47, K48, XXX, K4A, XXX, K4C, K4D }  \
+}
+
+#define LAYOUT_tsangan_ISO( \
```
Comment (fauxpark) [inline keyboards/han60/han60.h:75]:
```suggestion
#define LAYOUT_tsangan_iso( \
```

File: keyboards/han60/han60.h [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+	K41, K42, K43,                K44,                     K47, K49, K4B  \
+) { \
+    { K00, K01, K02, K03, K04, K05, K06, K07, K08, K09, K0A, K0B, K0C, K0D }, \
+    { K10, K11, K12, K13, K14, K15, K16, K17, K18, K19, K1A, K1B, K1C, K1D }, \
+	{ K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K2A, K2B, K2C, K2D }, \
+	{ K30, K31, K32, K33, K34, K35, K36, K37, K38, K39, K3A, K3B, XXX, XXX }, \
+	{ K40, K41, K42, K43, K44, XXX, XXX, K47, XXX, K49, XXX, K4B, XXX, XXX }  \
+}
+
+#define LAYOUT_tsangan_ISO_arrowkeys( \
```
Comment (fauxpark) [inline keyboards/han60/han60.h:89]:
```suggestion
#define LAYOUT_tsangan_iso_arrowkeys( \
```

File: keyboards/han60/info.json [hunk 15]
```diff
@@ -0,0 +1,341 @@
...
+                { "x": 1.5, "y": 4 },
+                { "w": 1.5, "x": 2.5, "y": 4 },
+                { "w": 7, "x": 4, "y": 4 },
+                { "x": 11, "y": 4 },
+                { "x": 12, "y": 4 },
+                { "x": 13, "y": 4 },
+                { "x": 14, "y": 4 }
+            ]
+        },
+        "LAYOUT_tsangan_ISO": {
```
Comment (fauxpark) [inline keyboards/han60/info.json:207]:
```suggestion
        "LAYOUT_tsangan_iso": {
```

File: keyboards/han60/info.json [hunk 16]
```diff
@@ -0,0 +1,341 @@
...
+                { "w": 1.5, "x": 0, "y": 4 },
+                { "x": 1.5, "y": 4 },
+                { "w": 1.5, "x": 2.5, "y": 4 },
+                { "w": 7, "x": 4, "y": 4 },
+                { "w": 1.5, "x": 11, "y": 4 },
+                { "x": 12.5, "y": 4 },
+                { "w": 1.5, "x": 13.5, "y": 4 }
+            ]
+        },
+        "LAYOUT_tsangan_ISO_arrowkeys": {
```
Comment (fauxpark) [inline keyboards/han60/info.json:272]:
```suggestion
        "LAYOUT_tsangan_iso_arrowkeys": {
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by farhandsome** [comment]
Hi there,

Just wanted to check in on this one.

**Comment by farhandsome** [comment]
@fauxpark @zvecr 

Hey all, just wanted to get an update on this!

File: keyboards/han60/rules.mk (hunk 1 above)
Comment (fauxpark) [reply keyboards/han60/rules.mk:21]: This is still outstanding.

**Review by fauxpark — CHANGES_REQUESTED** [review]
Please restore these files.

File: .gitignore [hunk 17]
```diff
@@ -86,3 +86,11 @@ __pycache__
...
 # Allow to exist but don't include it in the repo
 user_song_list.h
+keyboards/aos/tkl/keymaps/via/rules.mk
+keyboards/aos/tkl/keymaps/via/keymap.c
+keyboards/chavdai40/keymaps/42keys-dvorak/config.h
+keyboards/chavdai40/keymaps/42keys-dvorak/keymap.c
+keyboards/tempo_turtle/bradpad/keymaps/4brad/config.h
+keyboards/tempo_turtle/bradpad/keymaps/4brad/keymap.c
+keyboards/tempo_turtle/bradpad/keymaps/4brad/rules.mk
+lib/chibios-contrib
```
[the remark this answers is not in the document]
Comment (farhandsome) [reply .gitignore]: Oops, sorry about that.

**Review by fauxpark — CHANGES_REQUESTED** [review]
The files need to be restored too.

**Comment by farhandsome** [comment]
@fauxpark 

Do you know how I can restore the files? It won't let me revert the changes on Github Desktop.

**Comment by drashna** [comment]
You may need to rebase to fix the issues.

**Comment by farhandsome** [comment]
Closing this request and will re-open without the gitignore issues.

**Comment by farhandsome** [comment]
#15332 

This is now the current request for the Han60 files.

**Pull request closed without merging**